hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/freon/containergenerator/GeneratorDatanode.java

```diff
@@ -0,0 +1,317 @@
+import org.apache.hadoop.ozone.container.keyvalue.impl.ChunkManagerFactory;
+import org.apache.hadoop.ozone.container.keyvalue.interfaces.BlockManager;
+import org.apache.hadoop.ozone.container.keyvalue.interfaces.ChunkManager;
+
+import com.codahale.metrics.Timer;
+import picocli.CommandLine.Command;
+import picocli.CommandLine.Option;
+
+/**
+ * Container generator for datanode metadata/data.
+ */
+@Command(name = "cgdn",
+    description = "Offline container metadata generator for Ozone Datanodes.",
+    optionListHeading =
+        "\nExecute this command with different parameters for each datanodes. "
+            + "For example if you have 10 datanodes, use "
+            + "'ozone freon cgdn --index=1 --datanodes=10', 'ozone freon"
+            + " cgdn --index=2 --datanodes=10', 'ozone freon cgdn "
+            + "--index=3 --datanodes=10', ...\n\n",
+    versionProvider = HddsVersionProvider.class,
+    mixinStandardHelpOptions = true,
+    showDefaultValues = true)
+public class GeneratorDatanode extends BaseGenerator {
+
+  @Option(names = {"--datanodes"},
+      description = "Number of datanodes (to generate only subsequent of the "
+          + "required containers).",
+      defaultValue = "3")
+  private int datanodes;
+
+  @Option(names = {"--index"},
+      description = "Index of the datanode. For example datanode #3 should "
+          + "have only every 3rd container in a 10 node cluster.).",
+      defaultValue = "1")
+  private int datanodeIndex;
+
+  @Option(names = {"--zero"},
+      description = "User zero bytes instead of random data.",
+      defaultValue = "false")
+  private boolean zero;
+
+  private ChunkManager chunkManager;
+
+  private RoundRobinVolumeChoosingPolicy volumeChoosingPolicy;
+
+  private MutableVolumeSet volumeSet;
+
+  private Checksum checksum;
+
+  private ConfigurationSource config;
+
+  private Timer timer;
+
+  //Simulate ratis log index (incremented for each chunk write)
+  private int logCounter;
+  private String datanodeId;
+  private String scmId;
+  private int numberOfPipelines;
+  private int currentPipeline;
+
+  @Override
+  public Void call() throws Exception {
+    init();
+
+    numberOfPipelines = datanodes / 3;
+
+    //generate only containers for one datanodes
+    setTestNo(getTestNo() / numberOfPipelines);
+
+    currentPipeline = (datanodeIndex - 1) % numberOfPipelines;
+
+    config = createOzoneConfiguration();
+
+    BlockManager blockManager = new BlockManagerImpl(config);
+    chunkManager = ChunkManagerFactory
+        .createChunkManager(config, blockManager);
+
+    final Collection<String> storageDirs =
+        MutableVolumeSet.getDatanodeStorageDirs(config);
+
+    String firstStorageDir =
+        StorageLocation.parse(storageDirs.iterator().next())
+            .getUri().getPath();
+
+    final Path hddsDir = Paths.get(firstStorageDir, "hdds");
+    if (!Files.exists(hddsDir)) {
+      throw new NoSuchFieldException(hddsDir
```

Comment:
`NoSuchFieldException` seems strange in this context.  Intended `NoSuchFileException`?

Reply:
What about `NoSuchFieldNeiterAnyFileOrAnythingElseException`? ;-)

(Ok, just joking, I think It's just my usual auto-correct mistake. Anyway thanks for all the suggestions, will update the PR soon...)